hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneClientProducer.java

```diff
@@ -103,6 +107,13 @@ private OzoneClient getClient(OzoneConfiguration config) throws IOException {
           throw S3_AUTHINFO_CREATION_ERROR;
         }
 
+      } else {
```

Comment:
Should we move the createRemoteUser, setLoginUser outside the if condition and addToken when security is enabled?